gateway/internal/petition/petition_controller.go

```diff
@@ -28,32 +26,66 @@ func NewPetitionController(service IPetitionService) IPetitionController {
 }
 
 func (c *petitionController) CreatePetition(ctx *gin.Context) {
-	ctx.JSON(http.StatusOK, "nice")
+	var petition model.Petition
+	err := ctx.BindJSON(&petition)
+	if err != nil {
+		ctx.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+
+	resp, err := c.service.CreatePetition(petition)
+	if err != nil {
+		ctx.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
 
+	ctx.JSON(http.StatusCreated, resp)
 }
 
 func (c *petitionController) UpdatePetition(ctx *gin.Context) {
-	ctx.JSON(http.StatusOK, "nice")
+	var petition model.Petition
+	err := ctx.BindJSON(petition)
+	if err != nil {
+		ctx.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
 
-}
+	msg, err := c.service.UpdatePetition(petition.PetitionId, string(petition.Status))
+	if err != nil {
+		ctx.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
 
-func (c *petitionController) GetPetition(ctx *gin.Context) {
-	ctx.JSON(http.StatusOK, "nice")
+	ctx.JSON(http.StatusOK, msg)
 
 }
 
-func (c *petitionController) DeletePetition(ctx *gin.Context) {
-	ctx.JSON(http.StatusOK, "nice")
-}
+func (c *petitionController) GetPetitions(ctx *gin.Context) {
+	query := model.PaginationQuery{}
+	err := ctx.BindJSON(&query)
+	if err != nil {
+		ctx.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
 
-func (c *petitionController) GetAllPetitions(ctx *gin.Context) {
-	ctx.JSON(http.StatusOK, "nice")
-}
+	petitions, err := c.service.GetPetitions(query)
+	if err != nil {
+		ctx.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
+	}
+
+	ctx.JSON(http.StatusOK, petitions)
 
-func (c *petitionController) SignPetition(ctx *gin.Context) {
-	ctx.JSON(http.StatusOK, "nice")
 }
 
-func (c *petitionController) SetStatus(ctx *gin.Context) {
-	ctx.JSON(http.StatusOK, "nice")
+func (c *petitionController) DeletePetition(ctx *gin.Context) {
```

Comment:
Why not just get the petition id from the URL? Why are you binding a whole petition from the request body?

Reply:
I know, m-am condus după proto messages când am făcut, dar poate nu am înțeles corect. Voi corecta când vin la oficiu